addons/analytic/models/analytic_account.py

```diff
@@ -93,6 +136,41 @@ class AccountAnalyticLine(models.Model):
     def _default_user(self):
         return self.env.context.get('user_id', self.env.user.id)
 
+
+
+    def _compute_amount_user_currency(self):
+        """ Compute the amount into the user's currency
+        """
+        context = dict(self._context or {})
+        user_currency_id = self.env.user.company_id.currency_id
+        ctx = context.copy()
+        for line in self:
+            ctx['date'] = line.date
+            company_currency_id = line.company_id.currency_id
+            amount = line.amount
+            if user_currency_id == company_currency_id:
```

Comment:
That strikes me as a pointless optimisation, the only thing it potentially saves is a rounding (which you can disable with round=True anyway).